app/models/branch.rb

```diff
@@ -0,0 +1,7 @@
+class Branch < ApplicationRecord
+
+  has_many :reservations, dependent: :restrict_with_error
```

Comment:
why restrict_with_error? give reason

Reply:
Sir, we can not delete a branch, if it has reservations.
Deleting branch without reservations, we wouldn't know the branch of reservations.
Deleting reservations with branch, will result in deleting line items too and can delete important data for users.